ferretdb_packaging/defineversion/main_test.go

```diff
@@ -0,0 +1,251 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"testing"
+
+	"github.com/sethvargo/go-githubactions"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// getEnvFunc implements [os.Getenv] for testing.
+func getEnvFunc(t *testing.T, env map[string]string) func(string) string {
+	t.Helper()
+
+	return func(key string) string {
+		val, ok := env[key]
+		require.True(t, ok, "missing key %q", key)
+
+		return val
+	}
+}
+
+func TestDefine(t *testing.T) {
+	for name, tc := range map[string]struct {
+		env            map[string]string
+		defaultVersion string // pg_documentdb/documentdb.control file's default_version field
+		expected       string
+	}{
+		"pull_request": {
+			env: map[string]string{
+				"GITHUB_EVENT_NAME": "pull_request",
+				"GITHUB_HEAD_REF":   "define-docker-tag",
+				"GITHUB_REF_NAME":   "1/merge",
+				"GITHUB_REF_TYPE":   "branch",
+			},
+			defaultVersion: "0.100-0",
+			expected:       "0.100.0~pr~define~docker~tag",
+		},
+
+		"pull_request/dependabot": {
+			env: map[string]string{
+				"GITHUB_EVENT_NAME": "pull_request",
+				"GITHUB_HEAD_REF":   "dependabot/submodules/tests/mongo-go-driver-29d768e",
+				"GITHUB_REF_NAME":   "58/merge",
+				"GITHUB_REF_TYPE":   "branch",
+			},
+			defaultVersion: "0.100-0",
+			expected:       "0.100.0~pr~mongo~go~driver~29d768e",
+		},
+
+		"pull_request_target": {
+			env: map[string]string{
+				"GITHUB_EVENT_NAME": "pull_request_target",
+				"GITHUB_HEAD_REF":   "define-docker-tag",
+				"GITHUB_REF_NAME":   "main",
+				"GITHUB_REF_TYPE":   "branch",
+			},
+			defaultVersion: "0.100-0",
+			expected:       "0.100.0~pr~define~docker~tag",
+		},
+
+		"push/main": {
+			env: map[string]string{
+				"GITHUB_EVENT_NAME": "push",
+				"GITHUB_HEAD_REF":   "",
+				"GITHUB_REF_NAME":   "main",
+				"GITHUB_REF_TYPE":   "branch",
+			},
+			defaultVersion: "0.100-0",
+			expected:       "0.100.0~main",
+		},
```

Comment:
It will not run for the push to the `main` branch because this code will never be in the `main` branch. `main` branch tracks upstream as is